Assets/Scripts/Framework/Animation/AnimationController.cs

```diff
@@ -0,0 +1,36 @@
+using System.Linq;
+using UnityEngine;
+
+namespace Framework.Animation
+{
+    public abstract class AnimationController : MonoBehaviour
+    {
+        protected Animator p_animator;
+
+        protected virtual void Awake() => p_animator = GetComponent<Animator>();
+
+        protected virtual void PlayAnimation(string animationName)
+        {
+            if (IsValidAnimation(animationName));
+                p_animator.SetTrigger(animationName);
+        }
+    
+        protected bool IsValidAnimation(string animationName)
+        {
+            if (p_animator == null)
+            {
+                Debug.LogError("Animator component is not assigned.");
+                return false;
+            }
+
+            bool animationParameters = p_animator.parameters.Any(param => param.name == animationName);
```

Comment:
```suggestion
            bool animationParameters = p_animator.parameters.Any(animationParam => animationParam.name == animationName);
```
Vind deze naam niet okie dokie, heel vaag het zegd niet echt wat het is. Als je iets beters weet dan mag dat ook.